safe/impact_function_v4/impact_function.py

```diff
@@ -1337,3 +1356,121 @@ def style(self):
         # Let's style the aggregation and analysis layer.
         simple_polygon_without_brush(self.aggregation_impacted)
         simple_polygon_without_brush(self.analysis_impacted)
+
+    @property
+    def provenance(self):
+        """Helper method to gather provenance for exposure_impacted layer.
+
+        :returns: Dictionary that contains all provenance.
+        :rtype: dict
+        """
+        # Environment
+        self._provenance['host_name'] = gethostname()
+        self._provenance['user'] = getpass.getuser()
+        self._provenance['qgis_version'] = QGis.QGIS_VERSION
+        self._provenance['gdal_version'] = gdal.__version__
+        self._provenance['qt_version'] = QT_VERSION_STR
+        self._provenance['pyqt_version'] = PYQT_VERSION_STR
+        self._provenance['os'] = platform.version()
+        self._provenance['inasafe_version'] = get_version()
+
+        # InaSAFE
+        self._provenance['impact_function_name'] = self.name
+        self._provenance['impact_function_title'] = self.title
+        self._provenance['requested_extent'] = self.requested_extent
```

Comment:
I will just use aggregation_layer = None if it doesn't have aggregation layer. And the requested_extent will set to None.